### ISSUE TITLE
Adding Equations

### ISSUE THREAD

**Issue by Ragos81** [bennvan/dokuwiki-plugin-latexcaption#3]
It would be nice if the plugin could be also used for equations

**Comment by Ragos81** [comment]
@bennvan 
Sorry, for the late reply.

I changed the standard configuration of mathjax according the documentation from dokuwiki to show the automatic numbering, but without success.

```
MathJax.Hub.Config({
    tex2jax: {
        inlineMath: [ ["$","$"], ["\\(","\\)"] ],
        displayMath: [ ["$$","$$"], ["\\[","\\]"] ],
        processEscapes: true
    },
    TeX: { equationNumbers: {autoNumber: "AMS"} }
});
```

Mathjay - by the way - had it's last update on 2021-11-20, is based on mathjax v2 and it doesn't seem like the developer would have any intention to update to v3 [issue#37](https://github.com/liffiton/dokuwiki-plugin-mathjax/issues/37) at anytime. (Thats no judgement)

Even if it would have worked, how could I refer to a specific equation without giving it a unique ID, the number to that equation would change if another one is added before.

I know the caption should be on the right of the equation, and it's not the best solution to put a caption above or below it. But currently I don't see another solution. Wouldn't it also improve the workflow if captions and citations are always added in the same way, instead of using different approach for equations compared to figures or tables?

So I'm not really sure how to go ahead now.

**Comment by bennvan** [comment]
Yes I forgot I forked the mathjax repo and switched to v3 years ago. Feel free to install my fork in dokuwiki manually using the link: https://github.com/bennvan/dokuwiki-plugin-mathjax/archive/refs/heads/master.zip

Make sure the tags: 'ams' is in the config e.g.:
```
MathJax = {
    tex: {
        inlineMath: [ ["$","$"], ["\\(","\\)"] ],
        displayMath: [ ["$$","$$"], ["\\[","\\]"] ],
        processEscapes: true,
        tags: 'ams'
    },
    svg: {
        fontCache: "global",
    }
};
```

Write your equations in dokuwiki e.g.:
```
\begin{equation}
F = ma \label{mylabel1}
\end{equation}

\begin{equation}
E = mc^2 \label{mylabel2} 
\end{equation}

\begin{equation}
F = ma \tag{25} \label{mylabel3}
\end{equation}

No numbering:
\begin{equation*}
E = mc^2 
\end{equation*}
 
or

\begin{equation}
E = mc^2 \notag
\end{equation} 

Refer to equation \eqref{mylabel1}, \eqref{mylabel2} and \eqref{mylabel3}
```
Which renders as:
![image](https://github.com/user-attachments/assets/3a5ef670-f231-4bf1-a46e-bd325d3c92b5)

This is what you're after, right?

**Comment by Ragos81** [comment]
@bennvan 
Thanks a lot for the fast reply! Now, since you posted your equations, I found the issue with the numbering.
I used ``$$`` tags for my block equation, instead of begin and end

```
$$
x=\frac{-b\pm\sqrt{b^2-4ac}}{2a}
$$

\begin{equation}
x=\frac{-b\pm\sqrt{b^2-4ac}}{2a} \label{mylabel1}
\end{equation}

Reference to \eqref{mylabel1}

<codeblock |code_label>
<ltcaption>caption</ltcaption>
<code>
x=\frac{-b\pm\sqrt{b^2-4ac}}{2a}
</code>
</codeblock>

Reference to {{ref>code_label}}
```
The result looks the same, but without the automatic numbering.

![image](https://github.com/user-attachments/assets/0d0d100f-e454-43de-a03a-932a8e78c28b)

I'm still not a big fan of two different syntax for the reference, but at least I have the numbering and the chance to refer to it.

So I think that topic can be closed, but still would like to ask two more questions if you don't mind:  
Is there something similar to latex ``\autoref{mylabel1}`` function to automatically create a "Equation (1)" instead of "(1)"?
Could you give me a hint where I can change the font of the created reference to match the dokuwiki template?

**Comment by bennvan** [comment]
I am all for consistency, so I've done what I can to help with Mathjax referencing which also answers both of those questions. I planned to do something like this previously but never had the need yet and it seems very few people care about in-text referencing on a wiki. 

Updating to the latest code provides `mathjaxref (default:false)` and `alwaysautoref (default:true)` conf options. If you enable `mathjaxref` and disable `alwaysautoref` this syntax:

```
\begin{equation}
E = mc^2 \label{mylabel1}
\end{equation}

\begin{equation}
E = mc^2 \label{mylabel2}
\end{equation} 

<figure center|fig_label>
{{:wiki:dokuwiki-128.png?90|}}
<caption>caption</caption>
</figure>

Autoref equation: {{autoref>mylabel1}} Autoref figure: {{autoref>fig_label}}
Ref equation: {{ref>mylabel2}} Ref figure {{ref>fig_label}}
```

renders as:
![image](https://github.com/user-attachments/assets/1b72cf83-21fa-4e8f-82b9-13d0d022e18e)

To get the mathjax references to display properly you need to use chtml and set some config options. I have updated my mathjax fork with this working config and cdn as default.

**Comment by Ragos81** [comment]
Great, thanks for all the work!
I updated to your latest latexcaption + mathjax and changed the mathjax options _- as I assume the update didn't change to the defaults -_ to the following:

_plugin»mathjax»url_
`https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js`

_plugin»mathjax»config_
```
MathJax = {
    tex: {
        inlineMath: [ ["$","$"], ["\\\\(","\\\\)"] ],
        displayMath: [ ["$$","$$"], ["\\\\[","\\\\]"] ],
        processEscapes: true,
        tags: "ams"
    },
    chtml: {
        scale: 1,
        matchFontHeight: true, 
        mtextInheritFont: true,     
        merrorInheritFont: true
    }
};
```
_plugin»mathjax»asciimath_
``0``

_plugin»mathjax»configfile_ and _plugin»mathjax»mathtags_ are both empty.

Using your code (but changing ``<caption>caption</caption>`` to ``<ltcaption>caption</ltcaption>``) renders as

![Image](https://github.com/user-attachments/assets/644d215b-c36d-415b-aec1-5c9a164ddeef)

So now I have no clue where the issue comes from.
By the way, are the 4 backslashes in the mathjax config intended? I used only 2 before.

for reference, my previous mathjax config was

``https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js``
and
```
window.MathJax = {
  tex: {
    inlineMath: [ ["\\(","\\)"] ],
    displayMath: [ ["\\[","\\]"] ],
    processEscapes: true,
    tags: "ams",
    autoload: {
      color: [],
      colorv2: ['color']
    },
    packages: {'[+]': ['noerrors']}
  },
  options: {
    ignoreHtmlClass: 'tex2jax_ignore',
    processHtmlClass: 'tex2jax_process'
  },
  loader: {
    load: ['input/asciimath', '[tex]/noerrors']
  }
};
```

**Comment by bennvan** [comment]
> So now I have no clue where the issue comes from.

Did you enable the `mathjaxref` option on the latexcaption plugin? It is not enabled by default to pass references to javascript rendering since not all wiki installs use mathjax. 

> as I assume the update didn't change to the defaults

Settings won't change to the default included with plugins if it is found in `dokuwiki/conf/local.php`

> By the way, are the 4 backslashes in the mathjax config intended? I used only 2 before.

Confusing but they are escape characters. '\\' means something else in PHP so that's how the string needs to be formatted in the PHP file for it to be interpreted as 2 backslashes. In the dokuwiki admin interface, it should display as 2 backslashes as any string saved there is escaped automatically.

> for reference, my previous mathjax config was
    https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js

Either should be fine

**Comment by Ragos81** [comment]
> Did you enable the ``mathjaxref`` option on the latexcaption plugin? It is not enabled by default to pass references to javascript rendering since not all wiki installs use mathjax.

That was it! Now the references are shown as intended.

Thanks a lot for the support!
